server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexStateService.java

```diff
@@ -805,10 +827,26 @@ private void sendVerifyShardBlockRequest(
             final TransportVerifyShardIndexBlockAction.ShardRequest shardRequest = new TransportVerifyShardIndexBlockAction.ShardRequest(
                 shardId,
                 block,
+                true,
                 parentTaskId
             );
             shardRequest.timeout(request.ackTimeout());
-            client.executeLocally(TransportVerifyShardIndexBlockAction.TYPE, shardRequest, listener);
+            client.executeLocally(
+                TransportVerifyShardIndexBlockAction.TYPE,
+                shardRequest,
+                listener.delegateFailure((delegate, replicationResponse) -> {
+                    final TransportVerifyShardIndexBlockAction.ShardRequest req = new TransportVerifyShardIndexBlockAction.ShardRequest(
+                        shardId,
+                        block,
+                        false,
+                        parentTaskId
+                    );
```

Comment:
nit to make it more explicit this is phase 2:
```suggestion
                    final var phase2 = new TransportVerifyShardIndexBlockAction.ShardRequest(shardId, block, false, parentTaskId);
```

Reply:
Thanks, see 2b2f329b576f107efbc5480fc1a0030848b94079